app/scripts/controllers/dashboard.coffee

```diff
@@ -75,10 +80,44 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
         watchCompute 'allSchools',
           dependencies: ['viewMode', 'year', 'schoolType', 'rankBy', 'moreThan40']
           computer: ([viewMode, year, rest...]) ->
-            if year? and viewMode == 'schools' then loadSchools viewMode, year, rest...
+            if year? then loadSchools viewMode, year, rest...
             else
               null
 
+        watchCompute 'regions',
+          dependencies: ['viewMode']
+          computer: ([viewMode]) ->
+            if viewMode == 'regions' then loadRegions() else null
+
+        watchCompute 'detailedRegions',
+          dependencies: ['regions', 'allSchools']
+          waitForPromise: true
+          computer: ([regions, allSchools]) -> $q (resolve, reject) ->
+            unless regions? and allSchools?
+              resolve null
+            else
+              $q.all
+                  regions: regions
+                  schools: allSchools
+                .then ({regions, schools}) ->
+                  detailsByRegion = {}
+                  schoolsByRegion = groupBy schools, 'REGION'
+                  for region, schools of schoolsByRegion
+                    detailsByRegion[region] =
+                      # TODO: should these averages be weighted by number of pupils?
+                      AVG_GPA: averageProp schools, 'AVG_GPA'
```

Comment:
Yeah, they become `NaN` if the properties are missing, and later code checks schoolType so the right ones are used. Probably better to check and only assign the valid ones here though, I'll add it.